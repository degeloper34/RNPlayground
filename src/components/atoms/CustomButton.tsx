import {StyleSheet, Pressable, PressableProps} from "react-native";
import Colors from "../../constants/Colors";
import {CustomText} from "./CustomText";

interface IProps extends PressableProps {
  onPress?: () => void;
  text: string;
}

export function CustomButton({onPress, text}: IProps) {
  return (
    <Pressable onPress={onPress} style={styles.btnStyle}>
      <CustomText
        type={"bold"}
        fontSize={14}
        text={text}
        textColor={Colors.mainNight}
      />
    </Pressable>
  );
}

const styles = StyleSheet.create({
  btnStyle: {
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 30,
    backgroundColor: Colors.mainYellow,
    padding: 12,
  },
});
