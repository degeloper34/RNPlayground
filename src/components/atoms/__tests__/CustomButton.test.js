import renderer from "react-test-renderer";
import React from "react";
import {CustomButton} from "../CustomButton";

describe("CustomButton", () => {
  it("renders correctly with text", () => {
    const component = renderer
      .create(<CustomButton text={"btnText"} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly with onPress", () => {
    const onPress = jest.fn();
    const component = renderer
      .create(<CustomButton onPress={onPress} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it("renders correctly without text, onPress", () => {
    const component = renderer.create(<CustomButton />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
