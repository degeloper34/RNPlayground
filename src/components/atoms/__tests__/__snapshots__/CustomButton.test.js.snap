// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton renders correctly with onPress 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F1D00A",
      "borderRadius": 30,
      "justifyContent": "center",
      "padding": 12,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        undefined,
        Object {
          "color": "#21325E",
          "fontFamily": "sora-bold",
          "fontSize": 14,
          "textDecorationLine": "none",
        },
      ]
    }
  />
</View>
`;

exports[`CustomButton renders correctly with text 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F1D00A",
      "borderRadius": 30,
      "justifyContent": "center",
      "padding": 12,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        undefined,
        Object {
          "color": "#21325E",
          "fontFamily": "sora-bold",
          "fontSize": 14,
          "textDecorationLine": "none",
        },
      ]
    }
  >
    btnText
  </Text>
</View>
`;

exports[`CustomButton renders correctly without text 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F1D00A",
      "borderRadius": 30,
      "justifyContent": "center",
      "padding": 12,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        undefined,
        Object {
          "color": "#21325E",
          "fontFamily": "sora-bold",
          "fontSize": 14,
          "textDecorationLine": "none",
        },
      ]
    }
  />
</View>
`;

exports[`CustomButton renders correctly without text, onPress 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F1D00A",
      "borderRadius": 30,
      "justifyContent": "center",
      "padding": 12,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        undefined,
        Object {
          "color": "#21325E",
          "fontFamily": "sora-bold",
          "fontSize": 14,
          "textDecorationLine": "none",
        },
      ]
    }
  />
</View>
`;
